### PR TITLE
GCW-3132 Adding polymorphic relation changes

### PR DESCRIPTION
**Pull request by bharat619**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3132

This adds the polymorphic relationship changes as done in the API 
https://github.com/crossroads/api.goodcity/pull/999

### CODE REVIEW

File: addon/controllers/messages_base.js [hunk 1]
```diff
@@ -35,11 +41,13 @@ export default Ember.Controller.extend({
...
+    messages = this.get("isItemThread")
+      ? messages
+          .filterBy("messageableType", "Item")
+          .filterBy("messageableId", this.get("item.id"))
+      : messages
+          .filterBy("messageableType", "Offer")
+          .filterBy("messageableId", this.get("offer.id"));
     return messages.filter(m => {
       return Boolean(m.get("isPrivate")) === this.get("isPrivate");
     });
```
Comment (patrixr) [inline addon/controllers/messages_base.js:53]:
```suggestion
  messages: Ember.computed("allMessages.[]", "offer", "item", function() {
     const type = this.get("isItemThread") ? "item" : "offer";
     return this.get("allMessages")
          .filterBy("messageableType", _.capitalize(type))
          .filterBy("messageableId", this.get(`${type}.id`))
          .filter(m => Boolean(m.get("isPrivate")) === this.get("isPrivate"));
```

Comment (patrixr) [reply addon/controllers/messages_base.js:53]: another option would be to create computed properties for `recordType` and `recordId`

All optional suggestions btw

File: app/serializers/application.js [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+import { ActiveModelSerializer } from "active-model-adapter";
 
 // Polymorphic associations are not supported in ember-data beta version:
 // refer: https://github.com/emberjs/data/issues/1574
 
-export default ActiveModelSerializer.extend({
+function normalize(payload) {
+  const key =
+    (payload.hasOwnProperty("messages") && "messages") ||
+    (payload.hasOwnProperty("message") && "message");
```
Comment (patrixr) [inline app/serializers/application.js:9]: _technically_ nothing prevents the API from sending the `message` AND the `messages` keys.

Maybe something like `_.flatten([ payload['message'], payload['messages'] ]).filter(_.identity).forEach(() => do something)

**Comment by steveyken** [comment]
@bharat619 is there a reason this PR is still open?

**Comment by bharat619** [comment]
Closing this, as this is currently not needed for donor/admin app

**Pull request closed without merging**